### DISCUSSION TITLE
Table row break when page end.

### DISCUSSION THREAD

**Discussion by bhaloditarun** [QuestPDF/QuestPDF#564] [General]
Hello dear friends,

I'm working on generating table and rows using QuestPDF library there is one issue if anybody have solution for it then thanks in advance!!!

Issue is, table row is breaking in between when page is end. In other word half row content display in one page and half content is display on next page. **I just want to keep row content together(If have not enough space then whole row display in next page) .**

I have number of rows and content is dynamic so, can't predict how many rows display in one page.

There are some solutions like avoid-page-break: inside in other library but looking for solution if something for this library.

Thank you again,
Tarun

**Comment by MarcinZiabek** [comment]
Please use the `ShowEntire` element that prevents content from being split between page 😁

**Reply by bhaloditarun** [reply]
Thank you so much MarcinZiabek,
It works perfectly. 

Do we have any property for background opacity/transparency?

Actually, I'm using watermark image in page.

**Reply by MarcinZiabek** [reply]
For the background transparency/opacity, this approach may help https://github.com/QuestPDF/QuestPDF/discussions/569#discussioncomment-5997397 😁

**Reply by PlayerModu** [reply]
ShowEntire() literally just saved me so much pain. Thank you 👏

**Reply by MarcinZiabek** [reply]
Do you have any suggestions on how the API could be renamed to be more discoverable / more straightforward to find? :)

**Reply by PlayerModu** [reply]
I think the name of the API is fine once you know it exists and how to use it, as the context menu that comes up in Visual Studio when hovering does explain it further anyway :) Personally I think it would be worth calling it out explicitly in the QuestPDF Table documentation, maybe a new sub-header like "Row splitting over multiple pages" and then you could highlight your above comment solution to the problem?
